common/type_system/TypeSystem.cpp

```diff
@@ -70,6 +70,46 @@ Type* TypeSystem::add_type(const std::string& name, std::unique_ptr<Type> type)
   return m_types[name].get();
 }
 
+/*!
+ * Add a new 'enum type'. This maps enum names to the their type's name, allowing the enum name to
+ * be used as if it were a type name.
+ */
+Type* TypeSystem::add_enum_type(const std::string& name, const std::string& type) {
+  auto t = lookup_type(type);
+  assert(t);
+  auto kv = m_enum_types.find(name);
+  if (kv != m_enum_types.end()) {
+    // exists already
+
+    if (kv->second != type) {
+      // exists, and we are trying to change it!
+      fmt::print("[TypeSystem] Enum {} was originally\n{}\nand is redefined as\n{}\n", name,
+                 kv->second, type);
+
+      throw std::runtime_error("Enum type was redefined.");
+    }
+  } else {
+    // Should forward declaring enums be forbidden? would it even necessarily break anything? I'll
+    // keep it allowed for now, though it is a weird thing to do...
+    // if (m_forward_declared_types.find(name) != m_forward_declared_types.end()) {
+    //  fmt::print(
+    //      "[TypeSystem] Enum type {} was forward-declared, enums cannot be forward-declared\n");
+    //}
+
+    // newly defined!
+
+    if (m_types.find(t->get_parent()) == m_types.end()) {
```

Comment:
We can remove this check.  It's only needed when initially setting up the built-in types and all types should have valid parents by now.